allensdk/brain_observatory/filtered_running_speed/stim_file.py

```diff
@@ -0,0 +1,182 @@
+from typing import List, Union
+
+import pandas as pd
+import numpy as np
+
+
+class CamStimOnePickleStimFile(object):
+
+    @property
+    def data(self) -> dict:
+        return self._data
+
+    @property
+    def presentation_intervals(self) -> np.ndarray:
+        """Get an array of interval times (in ms) between stimulus
+        presentations
+        """
+        return np.array(self._data["intervalsms"])
+
+    @property
+    def num_frames(self) -> int:
+        """Get the number of stimulus frames presented during pkl session."""
+        return len(self.presentation_intervals) + 1
+
+    @property
+    def stimuli(self):
+        """List of dictionaries containing information about individual stimuli
+        """
+        return self._data["stimuli"]
+
+    @property
+    def frames_per_second(self):
+        """Framerate of stimulus presentation
+        """
+        return self._data["fps"]
+
+    @property
+    def pre_blank_sec(self):
+        """Time (s) before initial stimulus presentation
+        """
+        return self._data["pre_blank_sec"]
+
+    @property
+    def angular_wheel_velocity(self):
+        """ Extract the mean angular velocity of the running wheel
+        (degrees / s) for each frame.
+        """
+        return self.frames_per_second * self.angular_wheel_rotation
+
+    @property
+    def angular_wheel_rotation(self):
+        """ Extract the total rotation of the running wheel on each frame.
+        """
+        return self._extract_running_array("dx")
+
+    @property
+    def vsig(self):
+        """Running speed signal voltage
+        """
+        return self._extract_running_array("vsig")
+
+    @property
+    def vin(self):
+        return self._extract_running_array("vin")
+
+    def __init__(self, data, **kwargs):
```

Comment:
Please move the constructor to the top of the class

Reply:
Thanks. I've refactored my code to call the existing code.